### ISSUE TITLE
Error messages for things being the wrong item type should mention the real type!

### ISSUE THREAD

**Issue by guswynn** [MaterializeInc/materialize#15603]
For example, when we create a source with a name another item is using we get:
`ERROR:  catalog item 'thing' already exists`
and when we try to `DROP SOURCE` on an item that isnt a source we get:
`ERROR:  materialize.public.thing is not of type SOURCE`

Both could mention the type of the item `thing`

**Comment by chaas** [comment]
I added the type of the catalog item for errors like "is not of type x", which covers the second case in this ticket.

I looked into adding errors for the first type of error ("already exists"), but it's a bit more complicated b/c we never actually fetch the already-existing item, we just check for a name collision in a set. I'm not sure it's worth adding the plumbing to fetch the pre-existing item, but open to suggestions.

**Comment by guswynn** [comment]
Similarly, @aldeka mentioned that `ERROR:  unknown catalog item 'kafka_connection'` is unclear, as it doesn't mention its expecting a `CONNECTION`

**Comment by parker-timmerman** [comment]
@chaas I'm pretty sure this is fixed by https://github.com/MaterializeInc/materialize/pull/17889, is there anything from that PR that doesn't cover what's described in this issue though?

**Comment by guswynn** [comment]
@parker-timmerman does it cover the case where you do something like `DROP SOURCE <some item that isnt a source`? Right now that just says the item is NOT a source, but doesnt what kind of item it is

**Comment by parker-timmerman** [comment]
Ah shoot, no it doesn't do that, it only handles the cases where you try to create something named "foo", but "foo" already exists. In that case it'll tell you what the type of the already existing "foo" is

**Comment by parker-timmerman** [comment]
So in that case I won't close this, but I'll probably remove myself from assignee's, what do you think Cara?

**Comment by chaas** [comment]
Yeah that works. It's what I did with the first change back in November.
We can keep this in mind if you ever have downtime and are looking for something small to pick up.

**Comment by parker-timmerman** [comment]
Sounds great!

**Comment by chaas** [comment]
Example where the former case caused confusion in practice: https://materializeinc.slack.com/archives/CU7ELJ6E9/p1726477393463919
